lxd/device/proxy.go

```diff
@@ -181,13 +182,51 @@ func (d *proxy) validateEnvironment() error {
 	return nil
 }
 
+// validateNetworkForwardConflicts checks that the proxy device about to be created does not
+// overlap on existing network forward (both entities can't have the same listening address with
+// the same port number)
+func (d *proxy) validateNetworkForwardConflicts() error {
+	proxyListenAddr, err := ProxyParseAddr(d.config["listen"])
+	if err != nil {
+		return err
+	}
+
+	return d.state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+		projectNetworksForwardsOnUplink, err := tx.GetProjectNetworkForwardListenAddressesOnMember(ctx)
+		if err != nil {
+			return fmt.Errorf("Failed loading network forward listen addresses: %w", err)
+		}
+
+		for netFwdAddr, _ := range projectNetworksForwardsOnUplink {
+			if proxyListenAddr.Address == netFwdAddr {
+				return fmt.Errorf(
+					fmt.Sprintf(
+						"The proxy device ('%s') has an overlapping listen IP address ",
+						d.name,
+					) +
+					fmt.Sprintf(
+						"with an existing network forward IP address ('%s')",
+						netFwdAddr,
+					),
+				)
+			}
+		}
+		return nil
+	})
+}
+
 // Start is run when the device is added to the instance.
 func (d *proxy) Start() (*deviceConfig.RunConfig, error) {
 	err := d.validateEnvironment()
 	if err != nil {
 		return nil, err
 	}
 
+	err = d.validateNetworkForwardConflicts()
```

Comment:
Lets call this from `validateConfig` so it occurs at device creation/edit time as well as start time.

Also to avoid needing to parse the listen address multiple times with `ProxyParseAddr` can you update the `validateNetworkForwardConflicts` to accept a `net.IP` type that you can extract from `ProxyParseAddr` already called in `validateConfig` and then parse using `net.ParseIP`.